packages/fluentui/react-northstar/src/components/Alert/Alert.tsx

```diff
@@ -110,157 +127,211 @@ export const alertSlotClassNames: AlertSlotClassNames = {
-          defaultProps: () => ({
-            className: alertSlotClassNames.actions,
-            styles: styles.actions,
-          }),
+          defaultProps: () =>
+            getA11Props('actions', {
+              className: alertSlotClassNames.actions,
+              styles: resolvedStyles.actions,
+            }),
         })}
         {dismissible &&
           AlertDismissAction.create(dismissAction, {
-            defaultProps: () => ({
-              danger,
-              warning,
-              info,
-              success,
-              variables,
-              ...accessibility.attributes.dismissAction,
-            }),
-            overrideProps: this.handleDismissOverrides,
+            defaultProps: () =>
+              getA11Props('dismissAction', {
+                danger,
+                warning,
+                info,
+                success,
+                variables,
+              }),
+            overrideProps: handleDismissOverrides,
           })}
       </>
     );
   };
 
-  renderComponent(config: RenderResultConfig<AlertProps>) {
-    const { accessibility, classes, ElementType, unhandledProps } = config;
-    const { children } = this.props;
+  const element = (
+    <ElementType
+      {...getA11Props('root', {
+        className: classes.root,
+        onFocus: handleFocus,
+        ...unhandledProps,
+      })}
+    >
+      {childrenExist(children) ? children : renderContent()}
+    </ElementType>
+  );
+
+  setEnd();
+
+  return element;
+};
 
-    return (
-      <ElementType
-        className={classes.root}
-        onFocus={this.handleFocus}
-        {...accessibility.attributes.root}
-        {...rtlTextContainer.getAttributes({ forElements: [children] })}
-        {...unhandledProps}
-      >
-        {childrenExist(children) ? children : this.renderContent(config)}
-      </ElementType>
-    );
-  }
-}
+Alert.defaultProps = {
+  accessibility: alertBehavior,
+  dismissAction: {},
+  body: {},
+};
+
+Alert.propTypes = {
+  ...commonPropTypes.createCommon({ content: 'shorthand' }),
+  actions: PropTypes.oneOfType([customPropTypes.itemShorthand, customPropTypes.collectionShorthand]),
+  icon: customPropTypes.shorthandAllowingChildren,
+  header: customPropTypes.itemShorthand,
+  attached: PropTypes.oneOf([true, false, 'top', 'bottom']),
+  fitted: PropTypes.bool,
+  danger: PropTypes.bool,
+  defaultVisible: PropTypes.bool,
+  dismissible: PropTypes.bool,
+  dismissAction: customPropTypes.shorthandAllowingChildren,
+  info: PropTypes.bool,
+  onVisibleChange: PropTypes.func,
+  onFocus: PropTypes.func,
+  success: PropTypes.bool,
+  visible: PropTypes.bool,
+  warning: PropTypes.bool,
+  body: customPropTypes.shorthandAllowingChildren,
+};
+
+Alert.displayName = 'Alert';
+
+Alert.deprecated_className = alertClassName;
+
+Alert.handledProps = Object.keys(Alert.propTypes) as any;
+
+Alert.create = createShorthandFactory({
+  Component: Alert,
+});
 
+Alert.autoControlledProps = ['visible'];
```

Comment:
```suggestion
```

This can be removed

Reply:
If I remove this...

<img width="1367" alt="Screen Shot 2020-04-20 at 2 03 49 PM" src="https://user-images.githubusercontent.com/8545105/79749606-cd47bc00-830f-11ea-9c19-7445c2df9377.png">

visible appears as a prop on the panel on the right and the component here initially is hidden. How to solve then?

Reply:
There should not be dependency between these two. Can you please check `Alert.info.json`? Is a definition for `visible` is somehow different from `success` prop?

Reply:
```
{
      "description": "An alert can be set to visible to force itself to be shown.",
      "defaultValue": false,
      "tags": [],
      "types": [
        {
          "name": "boolean",
          "keyword": true
        }
      ],
      "name": "visible",
      "required": false
    },
```

```
{
      "description": "An alert can be formatted to display a successful message.",
      "defaultValue": false,
      "tags": [],
      "types": [
        {
          "name": "boolean",
          "keyword": true
        }
      ],
      "name": "success",
      "required": false
    },
```

Reply:
It's the same...

Reply:
Drop this to `packages/fluentui/docs/src/components/ComponentPlayground/componentGenerators.tsx`:

```tsx
export const Alert: KnobComponentGenerators<AlertProps> = {
  visible: ({ propName }) => ({
    hook: useBooleanKnob,
    name: propName,
    initialValue: true,
  }),
};
```

Reply:
Done